### PR TITLE
`Shw::Placeholder` - Fix color contrast

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

Context: https://github.com/hashicorp/design-system/pull/1695/files#r1347806114

### :hammer_and_wrench: Detailed description

In this PR I have:
- added default background color for `Shw::Placeholder`
- removed `@background` argument for `Shw::Placeholder` when the color was the same as the default
- replaced background colors in a couple of `Shw::Placeholder` instances

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/tests/dummy/app/styles/showcase-components/placeholder.scss [hunk 1]
```diff
@@ -18,6 +18,7 @@
   line-height: 1.2;
   text-align: center;
   text-shadow: 0 0 5px #fff;
+  background-color: #eee;
```
Comment (KristinLBradley) [inline packages/components/tests/dummy/app/styles/showcase-components/placeholder.scss:21]: Thanks for adding this. It makes the placeholder easier to use besides just fixing the contrast issue.

**Review by KristinLBradley — APPROVED** [review]

File: packages/components/tests/dummy/app/styles/showcase-components/placeholder.scss (hunk 1 above)
Comment (didoo) [reply packages/components/tests/dummy/app/styles/showcase-components/placeholder.scss:21]: I am not sure why I didn't think of it before 🤷‍♂️🤦‍♂️

**Comment by MelSumner** [comment]
Do you think we should have a few different conformant placeholder background colors? 

Looking at these changes through the lens of the side-nav page, if the background color is changed then there's not enough of a difference between the top background color and lower background color, see the two areas indicated here: 
CleanShot 2023-10-06 at 12 31 43

**Comment by didoo** [comment]
> Do you think we should have a few different conformant placeholder background colors?
> 
> Looking at these changes through the lens of the side-nav page, if the background color is changed then there's not enough of a difference between the top background color and lower background color, see the two areas indicated here: CleanShot 2023-10-06 at 12 31 43

Can you clarify what you suggest to do? I am not sure I understand

**Review by MelSumner — APPROVED** [review]

**Comment by MelSumner** [comment]
@didoo let's merge this and then I'll rebase that other PR and we can figure it out there.

**Comment by KristinLBradley** [comment]
> Do you think we should have a few different conformant placeholder background colors?
> 
> Looking at these changes through the lens of the side-nav page, if the background color is changed then there's not enough of a difference between the top background color and lower background color, see the two areas indicated here: CleanShot 2023-10-06 at 12 31 43

To me, it seems like that would overcomplicate usage of the Placeholder. I think we can rely on the a11y tests to catch other non-accessible color combinations when custom background colors are used.

**Pull request merged**